### PR TITLE
[DOCS] Document `nodes` cluster stats

### PR DESCRIPTION
**Pull request by jrodewig**
Documents the `nodes` response parameters returned by the
`_cluster/stats` API.

Also adds collapsible attributes for the `indices` and `nodes` sections.

### CODE REVIEW

File: docs/reference/cluster/stats.asciidoc [hunk 1]
```diff
@@ -338,7 +348,289 @@ This object is not populated by the cluster stats API.
...
+(integer)
+Number of selected nodes with the <<master-node,master-eligible>> role.
+
+`nodes.count.ml`::
+(integer)
+Number of selected nodes with the <<ml-node,machine learning>> role.
+
+`nodes.count.voting_only`::
+(integer)
+Number of selected nodes with the <<voting-only-node,voting-only>> role.
```
Comment (jbaiera) [inline docs/reference/cluster/stats.asciidoc]: Should we collapse these into one entry? The `nodes.count.coordinating_node` field is special in that it's for any nodes the have no role and will always be a valid entry, but roles are pluggable now so other values could end up here over time. 

If that feels like a bad idea from a docs perspective, then these are fine as-is.

File: docs/reference/cluster/stats.asciidoc [hunk 2]
```diff
@@ -338,7 +348,289 @@ This object is not populated by the cluster stats API.
...
+If no plugins or modules are installed, this array is empty.
+
+`nodes.plugins.name`:::
+(string)
+Name of the {es} plugin.
+
+`nodes.plugins.version`:::
+(string)
+{es} version for which the plugin was built.
+
```
Comment (jbaiera) [inline docs/reference/cluster/stats.asciidoc]: Missing `nodes.plugins.elasticsearch_version`
Also missing `node.plugins.java_version`

From Javadoc:

```
elasticsearchVersion  the version of Elasticsearch the plugin was built for
javaVersion           the version of Java the plugin was built with
```

File: docs/reference/cluster/stats.asciidoc [hunk 3]
```diff
@@ -338,7 +348,289 @@ This object is not populated by the cluster stats API.
...
+`nodes.plugins.description`:::
+(string)
+Short description of the plugin.
+
+`nodes.plugins.classname`:::
+(string)
+Class name used as the plugin's entry point.
+
```
Comment (jbaiera) [inline docs/reference/cluster/stats.asciidoc]: Missing `nodes.plugins.extended_plugins` typed `array[string]` which defines any other plugins this plugin extends through SPI, or empty array if it does not extend anything.

**Review by jbaiera — COMMENTED** [review]
Just a few items missing and one question on writing style

File: docs/reference/cluster/stats.asciidoc (hunk 1 above)
Comment (jrodewig) [reply docs/reference/cluster/stats.asciidoc]: This is a good idea. I left `coordinating_only` as-is but I used `nodes.count.<role>` for the remaining parms. Changes are in 5876819e1309b5d498a86fcfe2bd099be4aefc92.

File: docs/reference/cluster/stats.asciidoc (hunk 2 above)
Comment (jrodewig) [reply docs/reference/cluster/stats.asciidoc]: Added these with e47cf3fba015b381deca1990ff5a2913a11fedb0.

File: docs/reference/cluster/stats.asciidoc (hunk 3 above)
Comment (jrodewig) [reply docs/reference/cluster/stats.asciidoc]: Added this with 7d6a5e1cfa3dc301373ba3753a31d9ca20cafcef.

**Comment by jrodewig** [comment]
Thanks for the feedback, particularly catching those missing parms. I've made some changes to address. I'd appreciate a look at those when you're able:

* Node role changes: 5876819e1309b5d498a86fcfe2bd099be4aefc92
* Plugin versions: e47cf3fba015b381deca1990ff5a2913a11fedb0
* Extended plugins: 7d6a5e1cfa3dc301373ba3753a31d9ca20cafcef

**Review by jbaiera — APPROVED** [review]
LGTM!

**Comment by jrodewig** [comment]
Thanks @jbaiera!

**Comment by jrodewig** [comment]
### Backport commits
master 33d537f
7.x 641eb69520d
7.6 f09603d0222

**Pull request merged**